citesphere/src/main/java/edu/asu/diging/citesphere/core/service/giles/impl/GilesUploadCheckerImpl.java

```diff
@@ -219,4 +182,90 @@ public ICitation getCurrentCitation(ICitation citation, IUser user) {
         }
         return null;
     }
+    
+    private Set<IGilesUpload> getFileStatus(IGilesUpload upload, IUser user) {
+        Set<IGilesUpload> checkedUploads = new HashSet<>();
+        if (upload.getUploadingUser() == null
+                || Arrays.asList(GilesStatus.COMPLETE, GilesStatus.FAILED)
+                        .contains(upload.getDocumentStatus())) {
+            // in case something went wrong with the user
+            // or the upload has been processed
+            return null;
+        }
+        
+        if(user == null) {
+            user = userManager.findByUsername(upload.getUploadingUser());
+        }
+        String token = internalTokenManager.getAccessToken(user).getValue();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setBearerAuth(token);
+        HttpEntity<MultiValueMap<String, Object>> requestEntity = new HttpEntity<>(
+                headers);
+
+        ResponseEntity<String> response;
+        try {
+            response = restTemplate.exchange(
+                    gilesBaseurl + gilesCheckEndpoint + upload.getProgressId(),
+                    HttpMethod.GET, requestEntity, String.class);
+        } catch (HttpClientErrorException ex) {
+            upload.setDocumentStatus(GilesStatus.FAILED);
+            checkedUploads.add(upload);
+            return checkedUploads;
+        }
+        if (response.getStatusCode() == HttpStatus.ACCEPTED) {
+            // Giles is still procoessing
+            logger.debug("Upload " + upload.getProgressId()
+                    + " still being processed.");
+            checkedUploads.add(upload);
+        } else if (response.getStatusCode() == HttpStatus.OK) {
+            logger.debug("Upload " + upload.getProgressId() + " is done.");
+            ObjectMapper mapper = new ObjectMapper();
+            String jsonBody = response.getBody();
+            GilesUpload[] processed = new GilesUpload[0];
```

Comment:
is there a reason `processed` is initialized here? For what? either `readValue` will return a GilesUpload array, or the method will return.